### DISCUSSION TITLE
autocomplete api

### DISCUSSION THREAD

**Discussion by shaycomatrixdna** [ckan/ckan#7115] [General]
I am trying to use the autocomplete api (action/package_autocomplete) , but, i need it to work in Hebrew, as far as i can see the package_autocomplete method searches on the name and title fields of the dataset, but i could not find any way to enter them in Hebrew. 
Is there a way to change what auto complete is searching on ? can i include other fields in the auto complete?
any other solution?

thanks
Shay

**Comment by smotornyuk** [comment]
No, `package_autocomplete` is optimized for searching by title/name and does not have any way to customize its behavior. You can [create a new CKAN extension](https://docs.ckan.org/en/2.9/extensions/tutorial.html#creating-a-new-extension) that registers an API action, which does what you need. Here's a rough example of such action:
```python
 @tk.side_effect_free
 def package_autocomplete_by_field(context, data_dict):
     field, incomplete = tk.get_or_bust(data_dict, ["field", "incomplete"])

     # when searching by field, we should put quotes around term that contains spaces
     # or replace spaces with a wildcard
     pattern = incomplete.replace(" ", "?")
     result = tk.get_action("package_search")(context, {"q": f"{field}:*{pattern}*"})

     matches = []
     for package in result["results"]:
         if field in package:
             match_displayed = '%s (%s)' % (package['title'], package[field])
         else:
             match_displayed = package["title"]

         matches.append({
             'name': package['name'],
             'title': package['title'],
             'match_field': field,
             'match_displayed': match_displayed,
         })

     return  matches

```

Now you can call it like this:
```
http://ckan.domain/api/action/package_autocomplete_by_field?incomplete=cc&field=license_id
```
and it'll show you a couple of datasets, whose `license_id` contains `cc`. I suspect that you may need to configure properly [analyzer for the field](https://github.com/ckan/ckan/blob/master/ckan/config/solr/schema.xml#L52-L69) that contains terms in Hebrew because default CKAN's schema for Solr assumes English content.
